core/src/main/scala/sttp/tapir/Endpoint.scala

```diff
@@ -146,10 +147,17 @@ case class Endpoint[I, E, O, +S](input: EndpointInput[I], errorOutput: EndpointO
   def serverLogic[F[_]](f: I => F[Either[E, O]]): ServerEndpoint[I, E, O, S, F] = ServerEndpoint(this, f)
 }
 
-case class EndpointInfo(name: Option[String], summary: Option[String], description: Option[String], tags: Vector[String]) {
+case class EndpointInfo(
+    name: Option[String],
+    summary: Option[String],
+    description: Option[String],
+    tags: Vector[String],
+    deprecated: Option[Boolean]
```

Comment:
I'm wondering if this shouldn't be a simple `Boolean`. Because now we have three possible states: `Some(true)`, `Some(false)` and `None`. What's the difference betwee `Some(false)` and `None`?

Reply:
Mmh, I was coming from the openapi package and there I found same for other flags like `Parameter#required`, `Parameter#allowReserved` and so on. So, my expectation was, that it is wrapped in an `Option` to be able to exclude it from being encoded.

Reply:
Ah yes, in the `sttp.tapir.openapi` package classes have optional booleans - as that model reflects the openapi specification and what's later serialized as yaml/json. And there we indeed have 3 states: true, false, and "no value". How to interpret a "no value" scenario is up to the openapi specification.

However, in the higher-level, Scala representation, I think we shuold have only two possible states (either deprecated or not), with the default being `false` and visible in the code

Reply:
I see your point. The drawback would be to have `deprecated` set to `Some(false)` for most of the `OpenAPI` objects derived from `Endpoints`. Or I would need to translate a `false` set in higher-level into a `None` for `sttp.tapir.openapi` package. What would you prefer?

Reply:
I think the tapir->openapi interpreter should be aware of openapis defaults, and transalte `false -> None`, `true -> Some(true)`

Reply:
OK. I'll change it accordingly.